### PR TITLE
RSCBC-6: Add support for analytics

### PR DESCRIPTION
**Pull request by chvck**

### CODE REVIEW

File: sdk/couchbase/src/options/analytics_options.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    }
+}
+
+#[derive(Debug, Clone, TypedBuilder, Default)]
+#[builder(field_defaults(default, setter(into, strip_option)))]
+#[non_exhaustive]
+pub struct AnalyticsOptions<'a> {
+    pub client_context_id: Option<&'a str>,
+    pub priority: Option<i32>,
```
Comment (Matt-Woz) [inline sdk/couchbase/src/options/analytics_options.rs:28]: I think priority should be an `Option<bool>` in the public API?

File: sdk/couchbase-core/tests/query.rs [hunk 2]
```diff
@@ -110,6 +110,7 @@ async fn test_prepared_query_basic() {
     assert_eq!(0, meta.metrics.error_count);
     assert_eq!(0, meta.metrics.warning_count);
 
+    dbg!(&meta.signature);
```
Comment (Matt-Woz) [inline sdk/couchbase-core/tests/query.rs:113]: Leftover debug statement

File: sdk/couchbase/src/options/analytics_options.rs (hunk 1 above)
Comment (chvck) [reply sdk/couchbase/src/options/analytics_options.rs]: Good catch. Done.

File: sdk/couchbase-core/tests/query.rs (hunk 2 above)
Comment (chvck) [reply sdk/couchbase-core/tests/query.rs]: Done.

**Review by Matt-Woz — APPROVED** [review]

**Pull request merged**